src/opensles/StreamBuffered.cpp

```diff
@@ -44,17 +43,12 @@ Result StreamBuffered::open() {
         LOGD("StreamBuffered(): new FifoBuffer");
         mFifoBuffer = new FifoBuffer(getBytesPerFrame(), 1024); // TODO size?
         // Create a callback that reads from the FIFO
-        mInternalCallback = new AudioStreamBufferedCallback(this);
-        mStreamCallback = mInternalCallback;
-        LOGD("StreamBuffered(): mInternalCallback = %p", mInternalCallback);
+        mStreamCallback = std::make_shared<AudioStreamBufferedCallback>(this);
```

Comment:
This makes me nervous.  Can this object be deleted properly if it holds a reference to itself?

Reply:
Is "this" actually an AudioStreamBufferedCallback?

Reply:
"this" is the argument for AudioStreamBufferedCallback's constructor. Thus, the constructed callback will hold a raw pointer to the stream, while the stream will be holding a refcounted pointer to the callback.

Reply:
Yes, it's the equivalent of doing `new AudioStreamBufferedCallback(this)` except that instead of a raw pointer the result is a `shared_ptr` which will ensure deletion when the refcount drops to zero.

Reply:
Cool. Thanks for the explanation.